### PR TITLE
[fibrechannel] Add remote_ports for logging

### PR DESCRIPTION
**Pull request by darina-becker**
Add collecting information about remote ports

Resolves: #2284

Signed-off-by: Daria Bukharina d.bukharina@yadro.com

---
Please place an 'X' inside each '[]' to confirm you adhere to our [Contributor Guidelines](https://github.com/sosreport/sos/wiki/Contribution-Guidelines)

- [x] Is the commit message split over multiple lines and hard-wrapped at 72 characters?
- [x] Is the subject and message clear and concise?
- [x] Does the subject start with **[plugin_name]** if submitting a plugin patch or a **[section_name]** if part of the core sosreport code?
- [x] Does the commit contain a **Signed-off-by: First Lastname <email@example.com>**?
- [ ] If this commit closes an existing issue, is the line `Closes: #ISSUENUMBER` included in an independent line?
- [x] If this commit resolves an existing pull request, is the line `Resolves: #PRNUMBER` included in an independent line?

### CODE REVIEW

File: sos/report/plugins/fibrechannel.py [hunk 1]
```diff
@@ -17,7 +17,7 @@ class Fibrechannel(Plugin, RedHatPlugin):
 
     plugin_name = 'fibrechannel'
     profiles = ('hardware', 'storage', 'system')
-    files = ('/sys/class/fc_host')
+    files = ('/sys/class/fc_host', '/sys/class/fc_remote_ports')
```
Comment (TurboTurtle) [inline sos/report/plugins/fibrechannel.py:20]: The `files` tuple only affects plugin enablement, not what gets captured.

This plugin currently only captures `udevadm info` output for each fibre device identified during sos initialization. In particular that is handled here https://github.com/sosreport/sos/blob/59c5c461e36103d453cd9ef9b6cb4624a8171fbe/sos/report/__init__.py#L323

We do currently recurse through `/sys/class/fc_remote_ports` as part of our fibre device discovery.

Is there a case where a system using fibrechannel could potentially have the `fc_remote_ports` directory but _not_ the `fc_host` directory? Or is there information under this directory that we should be capturing separately from `udevadm`?

Comment (TurboTurtle) [reply sos/report/plugins/fibrechannel.py:20]: On second look, this does also fix a bug in the current plugin - `files` isn't a tuple currently, which is problematic.

Comment (darina-becker) [reply sos/report/plugins/fibrechannel.py:20]: @TurboTurtle Oh, sorry, you're right. I just ran it on the target machine before changes where remote ports are not avaliable and on the initiator after my additions that's why it looked working version! That was my fault sorry for taking your time.

Comment (TurboTurtle) [reply sos/report/plugins/fibrechannel.py:20]: Not at all :)

Like I said, this actually indirectly fixes a bug in the plugin right now. It certainly doesn't hurt anything by adding that path as an enablement check. I'm just not personally very familiar with fibrechannel so I wasn't/am not sure if it is possible to have a system with the `fc_remote_ports` path but not the `fc_host` path.

Comment (darina-becker) [reply sos/report/plugins/fibrechannel.py:20]: I asked my colleagues - The situation we have system with the `fc_remote_ports` path but not the `fc_host` path is unlikely except there is any critical bag in a kernel or drivers.

**Pull request closed without merging**